### PR TITLE
Indy NYM Transaction Script Added

### PR DESCRIPTION
**Pull request by himanshulalarya**
Signed-off-by: Lal <himanshu.a.lal@accenture.com>

### CODE REVIEW

File: platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+did import ./dids.txt
+did list
+
+did use Th7MpTaRZVRYnPiabds81Y
+
+pool create mysandbox gen_txn_file=./genesis_pool_config
+pool connect mysandbox
+pool list
+
+ledger nym did=AAAMpTaRZVRYnPiabds81Y verkey=8QhFxKxyaFsJy4CyxeYX34dFH8oWqyBv1P4HLQCsoeLy role=TRUST_ANCHOR
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt]: How can we determine the suceess / failure of this command?

File: platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+cd indy
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 68DB5E88
+sudo add-apt-repository "deb https://repo.sovrin.org/sdk/deb xenial stable"
+sudo apt-get update
+sudo apt-get install -y libindy
+
+sudo apt-get install -y indy-cli
+
+indy-cli generate_txn.txt
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh]: How would you parse the output from the get-nym? (e.g. try to get the verkey)

File: platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt (hunk 1 above)
Comment (himanshulalarya) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt]: If the txn is successful, it displays the success message in the console that NYM has been received along with the metadata.
If it fails (like wrong verkey or verkey modification by stewerd) error msg is displayed that NYM has  been rejected.

Comment (juraj-durech) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt]: and how are we able to automate this check?

Comment (himanshulalarya) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt]: When the script runs it open and external terminal and displays the corresponding msg if successful or failed

File: platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh (hunk 2 above)
Comment (himanshulalarya) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh]: using ledger get-nym did="" send=false, the created request will be printed and stored into the cli context as json object.

Comment (juraj-durech) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh]: send - (optional) Send the request to the Ledger (True by default). If false then created request will be printed and stored into CLI context.
That means we won't query the ledger at all...

File: platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt (hunk 1 above)
Comment (juraj-durech) [reply platforms/hyperledger-indy/scripts/indy_nym_txn/generate_txn.txt]: and how would you automate the check whether the script encountered an error?

File: platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+cd indy
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 68DB5E88
+sudo add-apt-repository "deb https://repo.sovrin.org/sdk/deb xenial stable"
+sudo apt-get update
+sudo apt-get install -y libindy
+
+sudo apt-get install -y indy-cli
+
+indy-cli generate_txn.txt >> result.txt
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/scripts/indy_nym_txn/pool-transaction.sh]: use just >, so that the result.txt is rewritten

**Pull request merged**